### PR TITLE
Azure - Migrate snapshot to a resource group

### PR DESCRIPTION
**Pull request by bathina2**
## Change Overview

The following PR enables Azure to accept a resource group to which a newly create snapshot can belong.

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [x] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [ ] :sunflower: Feature
- [ ] :world_map: Documentation
- [ ] :robot: Test

## Issues

- K10-7358

## Test Plan

- [x] :muscle: Manual
- [ ] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

File: pkg/blockstorage/azure/azuredisk.go [hunk 1]
```diff
@@ -225,7 +225,12 @@ func (s *AdStorage) SnapshotCopyWithArgs(ctx context.Context, from blockstorage.
 		},
 	}
 
-	result, err := s.azCli.SnapshotsClient.CreateOrUpdate(ctx, s.azCli.ResourceGroup, snapName, createSnap)
+	migrateResourceGroup := s.azCli.ResourceGroup
+	if val, ok := args[blockstorage.AzureMigrateStorageKey]; ok && val != "" {
```
Comment (onkarbhat) [inline pkg/blockstorage/azure/azuredisk.go:229]: This should be AzureMigrateResourceGroup instead of AzureMigrateStorageKey ?

Comment (bathina2) [reply pkg/blockstorage/azure/azuredisk.go]: you are very correct

File: pkg/blockstorage/azure/azuredisk.go [hunk 2]
```diff
@@ -225,7 +225,11 @@ func (s *AdStorage) SnapshotCopyWithArgs(ctx context.Context, from blockstorage.
 		},
 	}
 
-	result, err := s.azCli.SnapshotsClient.CreateOrUpdate(ctx, s.azCli.ResourceGroup, snapName, createSnap)
+	migrateResourceGroup := s.azCli.ResourceGroup
+	if val, ok := args[blockstorage.AzureMigrateResourceGroup]; ok && val != "" {
+		migrateResourceGroup = val
+	}
```
Comment (carlbraganza) [inline pkg/blockstorage/azure/azuredisk.go:231]: Please add to the function doc comment to describe this key.

File: pkg/blockstorage/helpers.go [hunk 3]
```diff
@@ -36,6 +36,7 @@ const (
 	AzureResurceMgrEndpoint    = "AZURE_RESOURCE_MANAGER_ENDPOINT"
 	AzureMigrateStorageAccount = "AZURE_MIGRATE_STORAGE_ACCOUNT_NAME"
 	AzureMigrateStorageKey     = "AZURE_MIGRATE_STORAGE_ACCOUNT_KEY"
+	AzureMigrateResourceGroup  = "AZURE_MIGRATE_RESOURCE_GROUP"
```
Comment (carlbraganza) [inline pkg/blockstorage/helpers.go:39]: Shouldn't this be referenced in `pkg/blockstorage/blockstorage_test.go`?

**Review by carlbraganza — CHANGES_REQUESTED** [review]
Missing reference from the existing UT that covers the other Azure env variables.

**Comment by bathina2** [comment]
> Missing reference from the existing UT that covers the other Azure env variables.

That test is skipped.

**Review by carlbraganza — APPROVED** [review]

**Pull request merged**